driver/modern_bpf/helpers/extract/extract_from_kernel.h

```diff
@@ -815,10 +815,36 @@ static __always_inline bool extract__exe_upper_layer(struct inode *inode)
 
 		bpf_probe_read_kernel(&upper_dentry, sizeof(upper_dentry), vfs_inode + inode_size);
 
-		if(upper_dentry)
+		if(!upper_dentry)
+		{
+			return false;
+		}
+
+		struct dentry *dentry = (struct dentry *)BPF_CORE_READ(exe_file, f_path.dentry);
+
+		unsigned int d_flags = BPF_CORE_READ(dentry, d_flags);
+		// DCACHE_DISCONNECTED = 0x20
+		bool disconnected = (d_flags & 0x20);
+		if(disconnected)
 		{
 			return true;
 		}
+
+		// In kernels >=6.5 d_fsdata represents an ovl_entry_flag.
+		unsigned long flags = (unsigned long)BPF_CORE_READ(dentry, d_fsdata);
+		if(bpf_core_field_exists(((struct ovl_entry*)0)->flags))
+		{
+			// kernel <6.5
+			struct ovl_entry *oe = (struct ovl_entry*)BPF_CORE_READ(dentry, d_fsdata);
+			flags = (unsigned long)BPF_CORE_READ(oe, flags);
+		}
+
+		unsigned long has_upper = (flags & (1U << (OVL_E_UPPER_ALIAS)));
```

Comment:
if we remove the new files we need to use the explicit value here instead of the enum, as you have done above
```c
// DCACHE_DISCONNECTED = 0x20
```